vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane.yaml

```diff
@@ -78,11 +78,19 @@ spec:
         - type: phy
           id: enp1s0
           macAddress:
+{% if IMAGE_OS != "Centos" %}
             fromHostInterface: enp1s0
+{% else %}
+            fromHostInterface: eth1
```

Comment:
that is surprising. I would expect it to be eth0 if it was using the eth* naming. However, you should consider the following: that interface name is coming from the introspection data. and introspection, whether you deploy Ubuntu or CentOS, is always done with IPA, so the naming will remain the same, so there should not be a need for this condition. The bottom line is that this should work without modifications, both for Ubuntu and CentOS.

Reply:
Updated

Reply:
> that is surprising. I would expect it to be eth0 if it was using the eth* naming. However, you should consider the following: that interface name is coming from the introspection data. and introspection, whether you deploy Ubuntu or CentOS, is always done with IPA, so the naming will remain the same, so there should not be a need for this condition. The bottom line is that this should work without modifications, both for Ubuntu and CentOS.

It means, there is no need for modification in [https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane.yaml#L81](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane.yaml#L81) or any other `fromHostInterface` ?

Reply:
@maelk the interface naming is not working properly, here are few outputs from cloud-init and the node itself: 
```
[   36.940155] virtio_net virtio1 enp2s0: renamed from eth1
[   36.982242] virtio_net virtio0 enp1s0: renamed from eth0
```

```
ci-info: ++++++++++++++++++++++++++++++++++++++Net device info+++++++++++++++++++++++++++++++++++++++
ci-info: +--------+------+-----------------------------+---------------+--------+-------------------+
ci-info: | Device |  Up  |           Address           |      Mask     | Scope  |     Hw-Address    |
ci-info: +--------+------+-----------------------------+---------------+--------+-------------------+
ci-info: |  eth0  | True | fe80::22c:a3ff:fef7:5384/64 |       .       |  link  | 00:2c:a3:f7:53:84 |
ci-info: |  eth1  | True |       192.168.111.100       | 255.255.255.0 | global | 00:2c:a3:f7:53:86 |
ci-info: |  eth1  | True | fe80::22c:a3ff:fef7:5386/64 |       .       |  link  | 00:2c:a3:f7:53:86 |
ci-info: |   lo   | True |          127.0.0.1          |   255.0.0.0   |  host  |         .         |
ci-info: |   lo   | True |           ::1/128           |       .       |  host  |         .         |
ci-info: +--------+------+-----------------------------+---------------+--------+-------------------+
ci-info: +++++++++++++++++++++++++++++++Route IPv4 info+++++++++++++++++++++++++++++++
ci-info: +-------+---------------+---------------+---------------+-----------+-------+
ci-info: | Route |  Destination  |    Gateway    |    Genmask    | Interface | Flags |
ci-info: +-------+---------------+---------------+---------------+-----------+-------+
ci-info: |   0   |    0.0.0.0    | 192.168.111.1 |    0.0.0.0    |    eth1   |   UG  |
ci-info: |   1   | 192.168.111.0 |    0.0.0.0    | 255.255.255.0 |    eth1   |   U   |
ci-info: +-------+---------------+---------------+---------------+-----------+-------+
ci-info: +++++++++++++++++++Route IPv6 info+++++++++++++++++++
ci-info: +-------+-------------+---------+-----------+-------+
ci-info: | Route | Destination | Gateway | Interface | Flags |
ci-info: +-------+-------------+---------+-----------+-------+
ci-info: |   1   |  fe80::/64  |    ::   |    eth0   |   U   |
ci-info: |   2   |  fe80::/64  |    ::   |    eth1   |   U   |
ci-info: |   4   |   ff00::/8  |    ::   |    eth0   |   U   |
ci-info: |   5   |   ff00::/8  |    ::   |    eth1   |   U   |
ci-info: +-------+-------------+---------+-----------+-------+
Cloud-init v. 18.5 running 'modules:config' at Fri, 04 Sep 2020 08:23:22 +0000. Up 58.79 seconds.
Error: Connection activation failed: No suitable device found for this connection (device eth0 not available because profile is not compatible with device (mismatching interface name)).
```

```
metal3@test1-lbtbj ~]$ ip link show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000
    link/ether 00:2c:a3:f7:53:84 brd ff:ff:ff:ff:ff:ff
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000
    link/ether 00:2c:a3:f7:53:86 brd ff:ff:ff:ff:ff:ff
4: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN mode DEFAULT group default 
    link/ether 02:42:8f:ce:8d:75 brd ff:ff:ff:ff:ff:ff
[metal3@test1-lbtbj ~]$ nmcli device status
DEVICE   TYPE      STATE      CONNECTION  
eth1     ethernet  connected  System eth1 
docker0  bridge    connected  docker0     
eth0     ethernet  connected  System eth0 
lo       loopback  unmanaged  --          
[metal3@test1-lbtbj ~]$ netstat -i
Kernel Interface table
Iface             MTU    RX-OK RX-ERR RX-DRP RX-OVR    TX-OK TX-ERR TX-DRP TX-OVR Flg
docker0          1500        0      0      0 0             0      0      0      0 BMU
eth0             1500     2856      0      0 0            23      0      0      0 BMRU
eth1             1500    44725      0    451 0         22800      0      0      0 BMRU
lo              65536    61764      0      0 0         61764      0      0      0 LRU

```
Are you sure interface names coming from introspection would override that coming from the image? Or then what is done wrong here because we can see that cloud-init says interface is renamed but that is not reflected in the node itself.

Reply:
Yes, it is not overriding it I guess or the conditioning is not in the right place